src/compiler/checker.ts

```diff
@@ -16392,10 +16392,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             // [...X[]] is equivalent to just X[]
             return readonly ? globalReadonlyArrayType : globalArrayType;
         }
-        const memberIds = mapDefined(namedMemberDeclarations, node => node ? getNodeId(node) : undefined);

```

Comment:
this would previously compute the same cache key segment for `[undefined, labelNode]` and `[labelNode, undefined]`